File: awa/cookbook/client-workers/java/src/main/java/com/slalom/awa/workflows/helloworld/activities/AddTimestampActivity.java
package com.slalom.awa.workflows.helloworld.activities;

import com.slalom.awa.workflows.helloworld.models.AddTimestampActivityInput;
import io.temporal.activity.ActivityInterface;
import io.temporal.activity.ActivityMethod;

@ActivityInterface
public interface AddTimestampActivity {

    @ActivityMethod(name = "add-timestamp-activity")
    String run(AddTimestampActivityInput input);
}

File: awa/cookbook/client-workers/dotnet/AgenticWorkflowApp.Console/Workflows/HelloWorld/Activities/AddTimestampActivity.cs
using AgenticWorkflowApp.Console.Workflows.HelloWorld.Models;
using Temporalio.Activities;

namespace AgenticWorkflowApp.Console.Workflows.HelloWorld.Activities;

public class AddTimestampActivity
{
    [Activity("add-timestamp-activity")]
    public Task<string> Run(AddTimestampActivityInput input) => Task.FromResult(string.Format(
            input.Format,
            input.Input,
            DateTime.Now));
}
